test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,171 @@
+	var (
+		oc = exutil.NewCLIWithPodSecurityLevel("gatewayapi-controller", admissionapi.LevelBaseline)
+	)
+	const (
+		// The expected OSSM subscription name.
+		expectedSubscriptionName = "servicemeshoperator3"
+		// The expected OSSM operator namespace.
+		expectedSubscriptionNamespace = "openshift-operators"
+		// The gatewayclass name used to create ossm and other gateway api resources.
+		gatewayClassName = "openshift-default"
+		// gatewayClassControllerName is the name that must be used to create a supported gatewayClass.
+		gatewayClassControllerName = "openshift.io/gateway-controller/v1"
+	)
+	g.BeforeEach(func() {
+		// create the default gatewayClass and confirm that it is accepted
+		gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+
+		gatewayClass := buildGatewayClass(gatewayClassName, gatewayClassControllerName)
+		gwc, err := gwapiClient.GatewayV1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
+		if err != nil {
+			e2e.Logf("Gateway Class %s already exists, or has failed to be created, checking its status", gwc.Name)
+		}
+
+		errCheck := checkGatewayClass(oc, gatewayClassName)
+		o.Expect(errCheck).NotTo(o.HaveOccurred())
+		e2e.Logf("GatewayClass %s successfully installed and accepted!", gwc.Name)
+
+	})
+
+	g.Describe("Verify Gateway API controller resources are created", func() {
+		g.It("and ensure OSSM related resources are created", func() {
+			coreClient := clientset.NewForConfigOrDie(oc.AdminConfig())
+			//check the subscription
+			g.By("Check OSSM subscription and CSV")
+			csvName, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "subscription", expectedSubscriptionName, "-o=jsonpath={.status.installedCSV}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			e2e.Logf("The subscription is installed and the CSV is: %v", csvName)
+			csvStatus, err := oc.AsAdmin().Run("get").Args("-n", expectedSubscriptionNamespace, "clusterserviceversion", csvName, "-o=jsonpath={.status.phase}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(csvStatus).To(o.Equal("Succeeded"))
+
+			g.By("check OSSM Pod and istiod pod is present with OSSM Operator installation")
+			podList, err := coreClient.CoreV1().Pods("openshift-operators").List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podList).To(o.ContainSubstring(`servicemesh-operator3`))
+
+			podList, err = coreClient.CoreV1().Pods("openshift-ingress").List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podList).To(o.ContainSubstring(`istiod-openshift-gateway`))
+
+			g.By("Confirm that ISTIO CR is created and in healthy state")
+			resource := types.NamespacedName{Namespace: "openshift-ingress", Name: "openshift-gateway"}
+
+			errIstio := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, false, func(context context.Context) (bool, error) {
```

Comment:
I don't think we need `wait.Poll` loop to wait the istio status is updated to `Healthy` in the test. Actually in BeforeEach `checkGatewayClass` function, the istio should already be in `Healthy` before the gatewayclass is accepted.